src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,924 @@
+
+      i_s3m1 = 3
+      decomp_cascade_con%cascade_step_name(i_s3m1) = 'S3M1'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s3m1) = rf_s3m1
+      cascade_donor_pool(i_s3m1) = i_soil3
+      cascade_receiver_pool(i_s3m1) = i_micr1
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s3m1) = 1.0_r8
+
+      i_l1m2 = 4
+      decomp_cascade_con%cascade_step_name(i_l1m2) = 'L1M2'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l1m2) = rf_l1m2
+      cascade_donor_pool(i_l1m2) = i_litr1
+      cascade_receiver_pool(i_l1m2) = i_micr2
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l1m2) = 1.0_r8
+
+      i_l2m2 = 5
+      decomp_cascade_con%cascade_step_name(i_l2m2) = 'L2M2'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l2m2) = rf_l2m2
+      cascade_donor_pool(i_l2m2) = i_litr2
+      cascade_receiver_pool(i_l2m2) = i_micr2
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_l2m2)= 1.0_r8
+
+      i_s3m2 = 6
+      decomp_cascade_con%cascade_step_name(i_s3m2) = 'S3M2'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s3m2) = rf_s3m2
+      cascade_donor_pool(i_s3m2) = i_soil3
+      cascade_receiver_pool(i_s3m2) = i_micr2
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s3m2) = 1.0_r8
+
+      ! TODO Should I have kept code corresponding to the s1_s3 s2_s3 transfers?
+      i_s1s3 = 7
+      decomp_cascade_con%cascade_step_name(i_s1s3) = 'S1S3'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s1s3) = rf_s1s3
+      cascade_donor_pool(i_s1s3) = i_soil1
+      cascade_receiver_pool(i_s1s3) = i_soil3
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s1s3) = 1.0_r8
+
+      i_s2s3 = 8
+      decomp_cascade_con%cascade_step_name(i_s2s3) = 'S2S3'
+      rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s2s3) = rf_s2s3
+      cascade_donor_pool(i_s2s3) = i_soil2
+      cascade_receiver_pool(i_s2s3) = i_soil3
+      pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_s2s3) = 1.0_r8
+
+      if (.not. use_fates) then
+         i_cwdl2 = 9
+         decomp_cascade_con%cascade_step_name(i_cwdl2) = 'CWDL2'
+         rf_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_cwdl2) = rf_cwdl2
+         cascade_donor_pool(i_cwdl2) = i_cwd
+         cascade_receiver_pool(i_cwdl2) = i_litr2
+         pathfrac_decomp_cascade(bounds%begc:bounds%endc,1:nlevdecomp,i_cwdl2) = cwd_flig
+      end if
+
+      deallocate(rf_s1s3)
+      deallocate(f_s1s3)
+
+    end associate
+
+  end subroutine init_decompcascade_mimics
+
+  !-----------------------------------------------------------------------
+  subroutine decomp_rates_mimics(bounds, num_soilc, filter_soilc, &
+       soilstate_inst, temperature_inst, ch4_inst, soilbiogeochem_carbonflux_inst)
+    !
+    ! !DESCRIPTION:
+    ! Calculate rate constants and decomposition pathways for the MIMICS
```

Comment:
I missed all these comments that still say "rate constants" in the latest commit. I marked them for the next commit :-)